src/dataflow/graph/graph.ts

```diff
@@ -177,6 +179,33 @@ export class DataflowGraph {
 		return this
 	}
 
+
+	/**
+	 * Adds a vertex for variable use. Intended for creating dataflow graphs as part of function tests.
+	 * @param id - AST node id
+	 * @param name - Variable name
+	 * @param when - always or maybe; defaults to always
+	 * @param environment - Environment the use occurs in
+	 * @param asRoot - boolean; defaults to true.
+	 * @param scope - string
+	 */
+	public uses(id: string, name: string, when: DataflowGraphEdgeAttribute = 'always', environment?: REnvironmentInformation, asRoot: boolean = true) {
```

Comment:
The core of these changes are helper methods like `uses` here, a specialized version of `addVertex` with less noise (no object syntax, explicit tags, etc.). Eventually, I plan to add specialized versions of all vertex types.

Reply:
Furthermore, these are many parameters. What about a mixed approach? 
`uses(id: NodeId, name: string, args: { when: ... })` this would save us from writing a lot of `undefine`s if the `asRoot` flag is to be set explicitly and it wouldn't create too much overhead (we could use a deep merge for the defaults or even, if we use a builder wrapper, configure those on-the-fly).